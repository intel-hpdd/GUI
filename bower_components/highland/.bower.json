{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}