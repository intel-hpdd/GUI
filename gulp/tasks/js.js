//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
