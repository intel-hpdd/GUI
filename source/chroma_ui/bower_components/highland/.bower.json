{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}