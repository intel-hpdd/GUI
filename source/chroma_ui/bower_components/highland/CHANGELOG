Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
