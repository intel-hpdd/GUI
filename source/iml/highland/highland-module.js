// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
