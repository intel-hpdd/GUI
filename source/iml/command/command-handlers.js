import * as fp from "@iml/fp";
import { render } from "inferno";
import getStore from "../store/get-store.js";
import getCommandStream from "./get-command-stream.js";
import global from "../global.js";
import socketStream from "../socket/socket-stream.js";
import multiStream from "../multi-stream.js";
import { setState, trimLogs } from "./command-transforms.js";

import { StepModalComponent } from "./step-modal.js";
import { CommandModalComponent } from "./command-modal.js";
import { querySelector } from "../dom-utils";
import { CLEAR_CONFIRM_ACTION } from "../action-dropdown/confirm-action-reducer.js";

const commandModalContainer = document.createElement("div");
const body = querySelector(global.document, "body");
getStore.select("commandModal").each(([...commands]: Command[]) => {
  const containerOnBody = body.contains(commandModalContainer);
  if (commands.length === 0 || containerOnBody === true) return;

  body.appendChild(commandModalContainer);

  const command$ = getCommandStream(commands);
  const onClose = () => {
    command$.end();
    render(null, commandModalContainer);
    body.removeChild(commandModalContainer);

    getStore.dispatch({
      type: CLEAR_CONFIRM_ACTION,
      payload: false
    });
  };

  render(<CommandModalComponent commands={[]} closeCb={onClose} />, commandModalContainer);
  command$
    .map(
      fp.map(
        fp.flow(
          trimLogs,
          setState
        )
      )
    )
    .each(([...xs]) => {
      render(<CommandModalComponent commands={xs} closeCb={onClose} />, commandModalContainer);
    });
});

const stepsModalContainer = document.createElement("div");
const extractApiId = fp.map(x => x.replace(/\/api\/step\/(\d+)\/$/, "$1"));
getStore.select("stepModal").each(({ ...job }: { job: JobT }) => {
  const containerOnBody = body.contains(stepsModalContainer);
  if (!job.id || containerOnBody === true) return;

  body.appendChild(stepsModalContainer);

  const stream = socketStream(`/job/${job.id}`);
  stream.write(job);

  const jobStream = stream.fork();
  jobStream.destroy = stream.destroy.bind(stream);

  const stepsStream1 = stream.fork();
  stepsStream1.destroy = stream.destroy.bind(stream);
  const stepsStream2 = stepsStream1
    .pluck("steps")
    .map(extractApiId)
    .flatMap(stepIds =>
      socketStream(
        "/step",
        {
          qs: {
            id__in: stepIds,
            limit: 0
          }
        },
        true
      )
    )
    .pluck("objects");
  stepsStream2.destroy = stepsStream1.destroy.bind(stream);

  const m$ = multiStream([jobStream, stepsStream2]);
  const onClose = () => {
    m$.end();
    render(null, stepsModalContainer);
    body.removeChild(stepsModalContainer);
  };

  render(<StepModalComponent job={job} steps={[]} closeCb={onClose} />, stepsModalContainer);
  m$.each(([job, steps]) => {
    render(<StepModalComponent job={job} steps={steps} closeCb={onClose} />, stepsModalContainer);
  });
});
