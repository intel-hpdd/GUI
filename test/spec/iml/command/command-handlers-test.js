// @flow

import highland from "highland";
import { render } from "inferno";

describe("command module", () => {
  let commandModal$, stepModal$, session$, modalStack$, socket$, getCommand$, container;
  let mockGetStore, mockGetCommandStream, mockSocketStream;

  beforeEach(() => {
    commandModal$ = highland();
    stepModal$ = highland();
    session$ = highland();
    modalStack$ = highland();

    mockGetStore = {
      select: jest.fn(name => {
        if (name === "commandModal") return commandModal$;
        else if (name === "stepModal") return stepModal$;
        else if (name === "session") return session$;
        else if (name === "modalStack") return modalStack$;
      }),
      dispatch: jest.fn()
    };
    jest.mock("../../../../source/iml/store/get-store.js", () => mockGetStore);

    getCommand$ = highland();
    mockGetCommandStream = jest.fn(() => getCommand$);
    jest.mock("../../../../source/iml/command/get-command-stream.js", () => mockGetCommandStream);

    socket$ = highland();
    mockSocketStream = jest.fn(() => socket$);
    jest.mock("../../../../source/iml/socket/socket-stream.js", () => mockSocketStream);

    require("../../../../source/iml/command/command-handlers.js");
  });

  describe("command modal", () => {
    let commands;

    beforeEach(() => {
      commands = [
        {
          cancelled: false,
          complete: true,
          created_at: "2019-01-01T00:00:00",
          errored: false,
          id: 7,
          jobs: ["/api/job/1"],
          logs: "logs",
          message: "message",
          state: "state",
          resource_uri: "resource_uri"
        }
      ];

      mockGetCommandStream.mockImplementation(() => getCommand$);
    });

    describe("with commands", () => {
      beforeEach(() => {
        commandModal$.write(commands);
        getCommand$.write(commands);

        if (document.body) container = document.body.querySelector("div");
      });

      afterEach(() => {
        render(null, container);
        if (document.body)
          while (document.body.children.length > 0) document.body.removeChild(document.body.children[0]);
      });

      it("should render the command modal", () => {
        expect(container).toMatchSnapshot();
      });

      describe("clicking the close button", () => {
        beforeEach(() => {
          let btn;
          if (container) btn = container.querySelector("button.btn-danger");
          if (btn) btn.click();
        });

        it("should close the modal", () => {
          let container;
          if (document.body) container = document.body.querySelector("div");

          expect(container).toBeNull();
        });

        it("should clear confirm action", () => {
          expect(mockGetStore.dispatch).toHaveBeenCalledWith({
            type: "CLEAR_CONFIRM_ACTION",
            payload: false
          });
        });
      });
    });

    describe("without commanmds", () => {
      it("should not render", () => {
        let container;
        commandModal$.write([]);

        if (document.body) container = document.body.querySelector("div");
        expect(container).toBeNull();
      });
    });
  });

  describe("step modal", () => {
    let job, steps;
    beforeEach(() => {
      job = {
        children: [],
        available_transitions: [
          {
            label: "Cancel",
            state: "cancelled"
          }
        ],
        cancelled: false,
        class_name: "StartTargetJob",
        commands: ["/api/command/230/"],
        created_at: "2019-03-05T17:23:14.773266",
        description: "Start target MGS",
        errored: false,
        id: 523,
        modified_at: "2019-03-05T17:23:14.773240",
        read_locks: [],
        resource_uri: "/api/job/523/",
        state: "tasked",
        step_results: {
          "/api/step/951/": null
        },
        steps: ["/api/step/951/", "/api/step/952/"],
        wait_for: [],
        write_locks: []
      };

      steps = [
        {
          state: "success",
          args: [],
          step_index: 1,
          step_count: 1,
          class_name: "RunAction",
          description: "RunAction bla",
          console: "console",
          backtrace: "backtrace"
        }
      ];

      mockSocketStream.mockImplementation(name => {
        if (name === "/job/523") return highland([job]);
        else if (name === "/step") return highland([{ objects: steps }]);
      });
    });

    describe("with data", () => {
      beforeEach(() => {
        stepModal$.write(job);

        if (document.body) container = document.body.querySelector("div");
      });

      afterEach(() => {
        render(null, container);
        if (document.body)
          while (document.body.children.length > 0) document.body.removeChild(document.body.children[0]);
      });

      it("should render the step modal", () => {
        expect(container).toMatchSnapshot();
      });

      describe("clicking the close button", () => {
        beforeEach(() => {
          let btn;
          if (document.body) btn = document.body.querySelector("button.btn-danger");
          if (btn) btn.click();
        });

        it("should close the modal", () => {
          let container;
          if (document.body) container = document.body.querySelector("div");

          expect(container).toBeNull();
        });
      });
    });

    describe("without a job id", () => {
      beforeEach(() => {
        delete job.id;
        stepModal$.write(job);
        if (document.body) container = document.body.querySelector("div");
      });

      it("should not render the modal", () => {
        expect(container).toBeNull();
      });
    });
  });
});
