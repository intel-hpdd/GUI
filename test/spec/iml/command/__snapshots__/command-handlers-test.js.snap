// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command module command modal with commands should render the command modal 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in command-modal"
      role="dialog"
      style="display: block; z-index: 1050;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Commands
            </h3>
          </div>
          <div
            class="modal-body command-modal-body"
          >
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="command-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  id="heading0"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-controls="collapse0"
                      aria-expanded="true"
                      data-parent="#command-accordion"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status fa-check"
                        />
                        <span>
                          message
                           - 
                          Jan 01 2019 00:00:00
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  aria-labelledby="heading0"
                  class="panel-collapse collapse in"
                  id="collapse0"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <h4>
                      Details:
                    </h4>
                    <table
                      class="table"
                    >
                      <tbody>
                        <tr>
                          <td>
                            Created At
                          </td>
                          <td>
                            Jan 01 2019 00:00:00
                          </td>
                        </tr>
                        <tr>
                          <td>
                            Status
                          </td>
                          <td>
                            Succeeded
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <div>
                      <h4>
                        Jobs
                      </h4>
                      <div
                        class="well jobs"
                      >
                        <div>
                          Loading Jobs... 
                          <i
                            class="fa fa-spinner fa-spin"
                          />
                        </div>
                      </div>
                    </div>
                    <div>
                      <h4>
                        Logs
                      </h4>
                      <pre
                        class="logs"
                      >
                        logs
                      </pre>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in command-modal-backdrop"
      style="z-index: 1049;"
    />
  </div>
</div>
`;

exports[`command module step modal with data should render the step modal 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in step-modal"
      role="dialog"
      style="display: block; z-index: 1060;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Start target MGS
            </h3>
          </div>
          <div
            class="modal-body "
          >
            <h4>
              Details:
            </h4>
            <table
              class="table"
            >
              <tbody>
                <tr>
                  <td>
                    Run At
                  </td>
                  <td>
                    Mar 05 2019 17:23:14
                  </td>
                </tr>
                <tr>
                  <td>
                    Status
                  </td>
                  <td>
                    <span
                      class="job-state"
                    >
                      <i
                        class="fa fa-refresh fa-spin"
                      />
                    </span>
                     
                    Running
                  </td>
                </tr>
              </tbody>
            </table>
            <h4>
              Steps
            </h4>
            
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="steps-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-expanded="true"
                      data-parent="#undefined"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status   fa-check "
                        />
                        <span>
                          2
                           / 
                          1
                           
                          RunAction
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  class="panel-collapse collapse in"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <div>
                      <h4>
                        Arguments
                      </h4>
                      <pre
                        class="logs"
                      >
                        []
                      </pre>
                      <div>
                        <h4>
                          Logs
                        </h4>
                        <pre
                          class="logs"
                        >
                          console
                        </pre>
                      </div>
                      <div>
                        <h4>
                          Backtrace
                        </h4>
                        <pre
                          class="logs"
                        >
                          backtrace
                        </pre>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in step-modal-backdrop"
      style="z-index: 1059;"
    />
  </div>
</div>
`;
